proposals/buildfile.md

```diff
@@ -0,0 +1,128 @@
+# Jib CLI Buildfile Specification
+
+Specification for a YAML buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```
+# "FROM" with option to use os/architecture for manifest lists
+baseImage: "gcr.io/distroless/java:8"
+baseImageOs: linux
+baseImageArchitecture: amd64
+
+creationTime: 0 # millis since epoch or iso8601 creation time
+format: Docker # Docker or OCI
+
+environment:
+  "MY_KEY1": "value"
+  "MY_KEY2": "value"
+labels:
+  "com.example.owner": "person"
+  "com.example.mode": "dev"
+volumes:
+  - "/myvolume"
+  - "/youvolume"
+
+exposedPorts:
+  - "123/udp"
+  - "456"
+  - "789/tcp"
+
+user: root
+workingDirectory: "/somewhere"
+entrypoint:
+  - "java"
+  - "-jar"
+cmd:
+  - "myjar.jar"
+
+# global file properties applied to all layers
```

Comment:
Including existing layers?  Maybe `layerDefaults`?

But maybe this is better handled by YAML aliases/anchors?  (Here's a [nice overview](https://medium.com/@kinghuang/docker-compose-anchors-aliases-extensions-a1e4105d70bd))

Reply:
It looks like anchors are kind of a "UI" thing, the parser should support it? This isn't something we have to implement in our data model?

Reply:
reworked this section